### PR TITLE
Added Randomisation Helper Methods

### PR DESCRIPTION
**Pull request by mmatic90**

### CODE REVIEW

File: .vs/ProjectSettings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "CurrentProjectSetting": "x64-Debug"
+}
```
Comment (kmlkcl) [inline .vs/ProjectSettings.json:3]: it seems we're pushing .vs folder . That shouldn't be happening. I'll update the .gitignore and remove those files from versioning.

File: CMakeLists.txt
Comment (kmlkcl) [inline CMakeLists.txt:1]: there are some changes on cmake list by @hukasu and it simplifies the cmake. Can you revert your cmake file and use his version? (Maybe you should add some new files etc but it's better in general)

File: framework/helper/randomisation.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#include "randomisation.h"
+#include "helper/logging.h"
+#include <time.h>
+#include <stdlib.h>
+#include <math.h>
+
+void T3_Init_Random(){
+    return srand(time(NULL));
+}
```
Comment (kmlkcl) [inline framework/helper/randomisation.cpp:9]: it's void but returns something? that is not a good idea. 

Also, this srand is what we use for randomization seed and I think it usually needs to be done only once (not sure though. needs testing) If that's the case, maybe we should just srand(time(NULL)); somewhere on our T3_Init method in main.cpp, and use no function for this one.

File: framework/helper/randomisation.cpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    T3_Assert(min>max, "Min value cannot be lower than Max.");
+
+    return min + rand() % (max - min + 1);
+}
+
+float T3_Random_Float(float min, float max){
+    T3_Assert(min>max, "Min value cannot be lower than Max.");
+
+    //Generate a random float between 0 and 1 with specified precision
+    float random_float_0_to_1 = ((float)rand() / RAND_MAX);
```
Comment (kmlkcl) [inline framework/helper/randomisation.cpp:29]: if something is between 0 and 1 we usually call it "normalized"

File: framework/helper/randomisation.h [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#ifndef T3_RANDOMISATION_H
+#define T3_RANDOMISATION_H
+
+void T3_Init_Random();                                      //Initialize the random generator seed (time)
+
+bool T3_Random_Bool();                                      //Generate a boolean random (returns true or false)
```
Comment (kmlkcl) [inline framework/helper/randomisation.h:6]: if you use this syntax above the method

```
/**
* Some explanation
*/ 
```
it also shows your explanation anywhere the method called. so maybe you can change the comments to this format?

File: main.cpp [hunk 5]
```diff
@@ -7,6 +7,11 @@
 #include "library/ecs/entity.h"
 #include "project/example-project-1/mushroom.h"
 
+//TEST
+#include <stdio.h>
+#include "helper/randomisation.h"
+//TEST END
+
```
Comment (kmlkcl) [inline main.cpp:14]: so you include the randomisation but didn't use anywhere ? :thinking:

**Review by kmlkcl — CHANGES_REQUESTED** [review]

File: .vs/ProjectSettings.json (hunk 1 above)
Comment (mmatic90) [reply .vs/ProjectSettings.json:3]: Okay, let me know if you need help with that, I won't touch that for now.

File: CMakeLists.txt
Comment (mmatic90) [reply CMakeLists.txt:1]: Okay, I switched to the reworked CMakeLists file. I didn't need to add any file or change anything.

File: framework/helper/randomisation.cpp (hunk 2 above)
Comment (mmatic90) [reply framework/helper/randomisation.cpp:9]: True that, I removed the T3_Init_Random() function.

srand is used to initialize the randomisation seed (using the time(NULL) as a reference, so it takes the seed from the current time). Every time the srand function is called a new "pseudo random" seed should be initialized, so it should not be called more than one time (reference: [c - srand() - StackOverflow](https://stackoverflow.com/questions/7343833/srand-why-call-it-only-once#:~:text=To%20avoid%20this%20problem%2C%20srand,different%20sequence%20of%20random%20numbers.)

I added the srand(time(NULL)) function in the main.cpp file, inside the T3_Init() method, so the seed gets randomised once the program is started.

File: framework/helper/randomisation.cpp (hunk 3 above)
Comment (mmatic90) [reply framework/helper/randomisation.cpp:29]: Refactored variable name in normalized.

P.S.: Thanks for the knowledge!

File: framework/helper/randomisation.h (hunk 4 above)
Comment (mmatic90) [reply framework/helper/randomisation.h:6]: Changed the comments syntax.

Thanks again, I did not know you can achieve that.

File: main.cpp (hunk 5 above)
Comment (mmatic90) [reply main.cpp:14]: My bad, remainder of tests. I got rid of those lines of code now.

**Review by kmlkcl — APPROVED** [review]
